### PR TITLE
Two flow errors fixed

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: src/selectors/entities/common.js [hunk 1]
```diff
@@ -13,7 +14,7 @@ export function getCurrentChannelId(state: GlobalState): string {
     return state.entities.channels.currentChannelId;
 }
 
-export function getMyChannelMemberships(state: GlobalState): string {
+export function getMyChannelMemberships(state: GlobalState): {[string]: ChannelMembership} {
```
Comment (saturninoabril) [inline src/selectors/entities/common.js]: Not sure if I understand this correctly, but why not `{string: ChannelMembership}`?

Comment (enahum) [reply src/selectors/entities/common.js]: It saying that it is an object with string values as properties and channelMembership as values

File: src/selectors/entities/common.js [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 import type {GlobalState} from '../../types/store';
 import type {UserProfile} from '../../types/users';
+import type {ChannelMembership} from '../../types/channels';
```
Comment (enahum) [inline src/selectors/entities/common.js]: Please change all these to be fullPath instead

Comment (enahum) [reply src/selectors/entities/common.js]: I know is not the scope of the PR but why not do it, right?

File: src/selectors/entities/common.js (hunk 1 above)
Comment (saturninoabril) [reply src/selectors/entities/common.js]: Thanks @enahum! Good to know, I initially thought that `[string]` is an "array of string values" but when I checked again array of strings is `Array<string>`.

Comment (jespino) [reply src/selectors/entities/common.js]: Yes, the problem is if you use {string: ChannelMembership} you are expressing that the objects have the key "string", and the value of type ChannelMemberships, in the other way you define "keys of type string" and values of type ChannelMemberships.

File: src/selectors/entities/common.js (hunk 2 above)
Comment (jespino) [reply src/selectors/entities/common.js]: Yes, the reason behind that is when you use it in other repositories, like in mattermost-webapp you find correctly it, if we use absolute path imports, it wont be found.

Comment (enahum) [reply src/selectors/entities/common.js]: Why?

Comment (enahum) [reply src/selectors/entities/common.js]: https://github.com/facebook/flow/issues/4186

Comment (jespino) [reply src/selectors/entities/common.js]: Yes, is a common problem. While I'm developing I can use the global reference, but the types sharing with other library is basically "copy" the file to export the types, so the absolute path in the other project doesn't work. I going to try to do something like "resolve locally mattermost-redux/whatever to src/whatever", allowing all imports use mattermost-redux/types/store. That would work for you?

Comment (jespino) [reply src/selectors/entities/common.js]: I uploaded a commit with the change, I have to test it in webapp flow changes, to verify that everything is working.

File: src/selectors/entities/common.js [hunk 3]
```diff
@@ -2,8 +2,9 @@
 // See License.txt for license information.
 // @flow
 
-import type {GlobalState} from '../../types/store';
-import type {UserProfile} from '../../types/users';
+import type {GlobalState} from 'mattermost-redux/types/store';
```
Comment (enahum) [inline src/selectors/entities/common.js]: So you are saying that `src/types/store` won’t work here?

This is sending mix signals to other developers, that is why I want us to be consistent

Comment (jespino) [reply src/selectors/entities/common.js]: Yeah, the problem is we have the compiled version, which simply compile to mattermost-redux/whatever, and the flow-types, which can be follow the same pattern from the outside, but can't follow the same pattern in the inside (with the current approach). We can rethink it, flow integration is a Work in Progress, so we have to polish it.

**Comment by jespino** [comment]
Already fixed in other PR

**Pull request closed without merging**